R/tax_unique.R

```diff
@@ -0,0 +1,279 @@
+    }
+  }
+
+  if (!is.null(paleobioDB)) {
+    paleobioDB$class <- gsub("NO_CLASS_SPECIFIED", NA, paleobioDB$class)
+    paleobioDB$order <- gsub("NO_ORDER_SPECIFIED", NA, paleobioDB$order)
+    paleobioDB$family <- gsub("NO_FAMILY_SPECIFIED", NA, paleobioDB$family)
+  }
+
+  if (!is.null(paleobioDB) &&
+      (any(grepl("[[:punct:]]", paleobioDB$class))) ||
+      (any(grepl("[[:punct:]]", paleobioDB$order))) ||
+      (any(grepl("[[:punct:]]", paleobioDB$family))) ||
+      (any(grepl("[[:punct:]]", paleobioDB$genus))) ||
+      (any(grepl("[[:punct:]]", paleobioDB$accepted_name)))) {
+    stop("paleobioDB taxonomy columns should not contain punctuation")
+  }
+
+  if ((!is.null(class) && !is.vector(class)) ||
+      (!is.null(order) && !is.vector(order)) ||
+      (!is.null(family) && !is.vector(family)) ||
+      (!is.null(genus) && !is.vector(genus)) ||
+      (!is.null(species) && !is.vector(species))) {
+    stop("Taxononic information must be in a vector")
+  }
+
+  if (!is.null(species)) {
+    if (length(species) != length(genus) ||
+        length(species) != length(family) ||
+        (!is.null(order) && (length(species) != length(order))) ||
+        (!is.null(class) && (length(species) != length(class)))) {
+      stop("Taxononic vectors must all be the same length")
+        }
+  }
+
+  if ((!is.null(class) && any(grepl("[[:punct:]]", class))) ||
+      (!is.null(order) && any(grepl("[[:punct:]]", order))) ||
+      (!is.null(family) && any(grepl("[[:punct:]]", family))) ||
+      (!is.null(genus) && any(grepl("[[:punct:]]", genus))) ||
+      (!is.null(species) && any(grepl("[[:punct:]]", species)))) {
+    stop("Taxonomy vectors should not contain punctuation")
+  }
+
+  if ((resolution != "species") && (resolution != "genera")) {
+    stop("Resolution must be species or genera")
+  }
+
+  if (!is.null(by) && !is.null(paleobioDB)) {
+    if (!by %in% colnames(paleobioDB)) {
+    stop("by must be a column name within the paleobioDB data frame")
+    }
+  }
+
+  if (!is.null(by) && !is.null(species)) {
+    if (!is.vector(by) || length(by) != length(species)) {
+    stop("When using taxonomic vectors, by should also be a vector of the same
+         length as the other vectors")
+    }
+  }
+
+#Run function
+
+  if (!is.null(paleobioDB)) {
+    #Filter paleobioDB necessary columns
+    occurrences <- paleobioDB[, c("class", "order", "family",
+                                "genus", "accepted_name")]
+
+    #If a category is provided, add it as a column
+    if (!is.null(by)) {
+      occurrences$category <- paleobioDB[[by]]
+    } else {
+      occurrences$category <- 1
+    }
+
+    #If accepted name is not a binomial, replace with NA
+    occurrences$accepted_name[grep(" ", occurrences$accepted_name,
+                                   invert = TRUE)] <- NA
+
+    #Rename column
+    colnames(occurrences)[colnames(occurrences) == "accepted_name"] <-
+      "genus_species"
+  } else
+
+  if (is.null(paleobioDB)) {
```

Comment:
Probably else if is not needed here? else should be sufficient if the data entry was not palaeobioDB?